system/Config/Routes.php

```diff
@@ -49,5 +49,11 @@
 $routes->cli('migrations/(:segment)', '\CodeIgniter\Commands\MigrationsCommand::$1');
 $routes->cli('migrations', '\CodeIgniter\Commands\MigrationsCommand::index');
 
-// CLI Catchall - uses a _remap to
+// CLI Catchall - uses a _remap to call Commands
 $routes->cli('ci(:any)', '\CodeIgniter\CLI\CommandRunner::index/$1');
+
+// Prevent access to BaseController
+$routes->add('basecontroller(:any)', function()
```

Comment:
Shouldn't this be basecontroller/(:any)? 
It also seems that it should be in front of the CLI routing

Reply:
When I tested it if I included the slash then "https://example.com/basecontroller" was still a routable address - making it basically "basecontroller*" fixes that and would also cover "BaseController2", etc, if users added more.
As to whether it should appear before or after CLI that I don't know! There were no routes in the system config except the CLI ones so I wasn't sure where to put it.